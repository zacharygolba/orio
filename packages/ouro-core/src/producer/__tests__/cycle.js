// @flow

import Cycle from '../cycle'

const SOURCE = 'test'
let producer

beforeEach(() => {
  producer = new Cycle(SOURCE)
})

test('#constructor()', () => {
  expect(producer).toMatchSnapshot()
})

test('#@@iterator()', () => {
  let i = 0

  for (const item of producer) {
    if (i > SOURCE.length * 3) {
      break
    }

    expect(SOURCE).toContain(item)
    i += 1
  }
})
