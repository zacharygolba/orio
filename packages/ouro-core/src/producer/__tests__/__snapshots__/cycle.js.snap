// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#constructor() 1`] = `
Cycle {
  "done": false,
  "source": "test",
  "state": 0,
}
`;

exports[`#drop() 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`#drop() 2`] = `
Cycle {
  "done": true,
  "source": Array [],
  "state": 0,
}
`;
